### PR TITLE
drivers: gpio: Add polarity support for sx1509b

### PR DESCRIPTION
**Pull request by traavnik**
Add polarity support for sx1509b GPIO extender.
Most of the code was already in place, but I2C write was not implemented for some reason.
Tested on Nordic Thingy:52 (nrf52_pca20020)

Signed-off-by: Aleš Travnik <traavnik@outlook.com>

### CODE REVIEW

**Review by mbolivar — APPROVED** [review]
I don't know this hardware, but there's nothing scary in here.

Looks like @tkln originally added the driver; can you comment?

Also cc @mnkp

**Review by mnkp — APPROVED** [review]

File: drivers/gpio/gpio_sx1509b.c
Comment (nashif) [inline drivers/gpio/gpio_sx1509b.c]: use brackets around if statement

**Review by nashif — CHANGES_REQUESTED** [review]

File: drivers/gpio/gpio_sx1509b.c
Comment (traavnik) [reply drivers/gpio/gpio_sx1509b.c]: I fixed it but kind of made the pull request messy. Maybe there's a way to squash it before merge or should I create a new pull request?

Comment (mnkp) [reply drivers/gpio/gpio_sx1509b.c]: There is no need to create a new PR. We work by rewriting local history on a branch from which PR was created and force pushing the branch to Github. Some more info here [Contribution Workflow](https://docs.zephyrproject.org/latest/contribute/index.html#contribution-workflow).

The coding style requires us to use braces even for single-line code blocks. Since `gpio_sx1509b.c` already violates this rule in many places the proper way to issue this PR would be to submit it as two commits. The first one should fix all the existing violations to the 'braces' policy and the second should add support for polarity with the correct usage of braces.

Comment (traavnik) [reply drivers/gpio/gpio_sx1509b.c]: Thanks. I'm still trying to wrap my head around the workflow.

I understand. Conveniently others fixed the braces after my PR. I merged the changes with my branch before fixing the PR. Although, am I right to assume that updating `gpio_sx1509b.c` in my branch wasn't really needed in order to merge PR properly?

Comment (mnkp) [reply drivers/gpio/gpio_sx1509b.c]: Since others have already fixed remaining brackets, indeed you'll only need to push one commit containing your original changes. But in this specific case they still need to be applied on the latest version of the file or otherwise there would be merge conflicts. The other and your commit are modifying overlapping lines of code.

There are different ways to handle your problem, I would probably do it this way:
```
git checkout sx1509b
git reset master
```
assuming sx1509b tracks master branch. At this point you should only see your modifications and `git status` should say:
```
On branch sx1509b
Your branch is up to date with 'master'.
```
All your history for the last 3 commits is lost. You will need to re-type your original commit message. Now:
```
git add drivers/gpio/gpio_sx1509b.c
git commit -s
git push -f origin sx1509b
```

In the future, if you are faced with similar problem don't use `git merge` but rather `git rebase`. By default `git rebase` will put whatever work you have done on the latest version of the branch you track. In the process you may need to resolve merge conflicts.

Comment (traavnik) [reply drivers/gpio/gpio_sx1509b.c]: Thank you very much for your detailed explanation!
I hope this PR is now suitable for merging.

Comment (mnkp) [reply drivers/gpio/gpio_sx1509b.c]: Looks good, thanks! Now we need to wait for @nashif to merge it.

**Comment by galak** [comment]
@pabigot don't know if this will be addressed when you update the driver.

**Comment by pabigot** [comment]
The updated driver supports polarity, but not using the SX1509B hardware.

I was told that the original motivation for polarity control was to support what we're now handling with the physical vs logic level distinction per pins.  That's entirely done in software, and the existing polarity flags are deprecated in the new API.

This means that the hardware polarity feature cannot be controlled with the generic GPIO API.  If it's necessary/useful a driver-specific extension could be added, at the risk of confusion due to mixing software and hardware polarity features.

So I'm going to close this, and ask that a new issue be re-opened if there's reason to use hardware control.

**Pull request closed without merging**
